### PR TITLE
Backport PR #19108 on branch v3.3.x

### PR DESCRIPTION
**Pull request by timhoffm**
Backport PR #19108 on branch v3.3.x

In contrast to the original PR, no changes in tests are needed because we did not have tests that would trigger the warning back in 3.3.x.

### CODE REVIEW

**Review by QuLogic — DISMISSED** [review]

**Comment by QuLogic** [comment]
Are you sure about that? The test is still failing.

**Comment by tacaswell** [comment]
@QuLogic and I talked through this today.  The issue is:

 - The HTMLWriter inherits from FileMovieWritter which sets up a non-context manager TemporaryDirectory in the in init
 - However the cleanup code is not called (because the super class cleanup code does things that do not make sense for HTMLWriter)
 - we do not see this failure on the default branch because sometime after we branched v3.3.x off the tests got cleaned up a best and the object that triggers the failure is now created in a fixture so is escaping this pytest logic that converts un-handled exceptions into failures.

This code probably should on the main branch, we will handle that with a careful merge up.

**Comment by QuLogic** [comment]
It looks like there's another failure in pgf due to the `LatexManager` teardown. There is a weakref finalizer for the `Popen` added by @anntzer in #18679, but it doesn't appear to have been called early enough.

**Comment by anntzer** [comment]
#18679 doesn't appear to have been backported, and I guess it basically fixed the bug that is now getting exposed by pytest?

**Comment by tacaswell** [comment]
I tried to backport the non-deprecation related changes.

I think I missed these failures locally because I have other pgf failures due to latex changing under us (I see this on Arch, assume we will see it in test containers eventually, but that is a problem for the future).

**Review by tacaswell — APPROVED** [review]
I have commits in here so leave it to the judgement of further reviewers if they they feel comfortable merging or wait for additional review.

**Comment by QuLogic** [comment]
Ah, sorry, I was looking at `master` instead of `v3.3.x`.

File: lib/matplotlib/backends/backend_pgf.py
Comment (QuLogic) [inline lib/matplotlib/backends/backend_pgf.py:766]: This change was only for the deprecation, so the renaming isn't really needed here.

**Review by QuLogic — APPROVED** [review]

**Pull request merged**
